mlir/lib/Dialect/MIOpen/Pipelines/XMIRPipelines.cpp

```diff
@@ -21,26 +21,13 @@
 //===----------------------------------------------------------------------===//
 
 #include "mlir/Dialect/MIOpen/XMIRPipelines.h"
-#include "mlir/Dialect/MIOpen/Pipelines.h"
 
-#include "mlir/Conversion/MIOpenPasses.h"
-#include "mlir/Conversion/SCFToControlFlow/SCFToControlFlow.h"
-#include "mlir/Dialect/Arithmetic/Transforms/Passes.h"
-#include "mlir/Dialect/Async/Passes.h"
 #include "mlir/Dialect/MIOpen/Passes.h"
-#include "mlir/Dialect/Tensor/Transforms/Passes.h"
-#include "mlir/IR/Builders.h"
-#include "mlir/IR/BuiltinOps.h"
+#include "mlir/InitAllDialects.h"
```

Comment:
Is there any particular reason we're going for `InitAllDialects` here instead of including references to just the stuff we need? I could've sworn `InitAllDialects` was a convenience header for stuff like `mlir-opt` and that you weren't supposed to run around including it

Reply:
Could probably speed up compile a few seconds. If that gets to be a problem we should run a header minimizer on all the code.

Reply:
Yeah, I've been vaguely wanting to run a header minimizer for a while...